src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractLocalFunctionTests.cs

```diff
@@ -4851,5 +4851,36 @@ static void L()
     }
 }", codeActionIndex: 1);
         }
+
+        [WorkItem(55031, "https://github.com/dotnet/roslyn/issues/55031")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsExtractLocalFunction)]
+        public async Task ExtractLocalFunctionVariableNameShadowing_PriorCSharp8()
+        {
+            var code = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        var f = ""{0}"";
+        [|string.Format(f, """");|]
+    }
+}";
+            var expected = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        var f = ""{0}"";
+        {|Rename:NewMethod|}(f);
+
+        void NewMethod(string f)
```

Comment:
`f` is causing CS0136 in C# 7.3. The refactoring should either name the parameter differently or capture f.